### DISCUSSION TITLE
Suggestion: Add an Attribute to Disable Script Preloading in NextJS

### DISCUSSION THREAD

**Discussion by searchstefano** [vercel/next.js#66297] [App Router]
Hello everyone,

I hope you are doing well. Recently, I encountered the issue of inserting scripts within NextJS. After running various tests, I found that when using the “afterInteraction” strategy, NextJS inserts the resource among the preloads in the page head.

https://nextjs.org/docs/app/building-your-application/optimizing/scripts

Now, this is great for internal resources or external layout scripts, but in the case of using external scripts like ads, which you don't want to preload but still want to load after interaction, how should one proceed?

Wouldn't it be useful to add an attribute to the <Script> component that can disable resource preloading?

Thank you

**Comment by ionetek** [comment]
I have the same problem. At the moment, the only way is to use lazyOnload
